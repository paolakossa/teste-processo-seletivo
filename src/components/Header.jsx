import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Dropdown, Menu } from 'antd';
import { DownOutlined } from '@ant-design/icons';
import styled from 'styled-components';
import { logout } from '../store/auth/authSlice';

const HeaderContainer = styled.header`
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: #fff;
  padding: 0 1rem;
  border-bottom: 1px solid #ddd;
  height: 64px;
`;

const CompanyName = styled.div`
  font-weight: bold;
  font-size: 1.2rem;
`;

const UserContainer = styled.div`
  display: flex;
  align-items: center;
  gap: 8px;
  cursor: pointer;
`;

const RotatingIcon = styled(DownOutlined)`
  transition: transform 0.3s ease-in-out;
  ${({ isOpen }) => isOpen && 'transform: rotate(180deg);'}
`;

export const Header = () => {
  const dispatch = useDispatch();
  const email = useSelector((state) => state.userData.email);
  const [menuOpen, setMenuOpen] = useState(false);

  const menu = (
    <Menu
      items={[
        {
          key: 'logout',
          label: 'Sair',
          onClick: () => dispatch(logout()),
        },
      ]}
    />
  );

  return (
    <HeaderContainer>
      <CompanyName>ITA Frotas</CompanyName>
      <Dropdown overlay={menu} trigger={['click']} onOpenChange={(open) => setMenuOpen(open)}>
        <UserContainer>
          <span>{email}</span>
          <RotatingIcon isOpen={menuOpen} />
        </UserContainer>
      </Dropdown>
    </HeaderContainer>
  );
};
