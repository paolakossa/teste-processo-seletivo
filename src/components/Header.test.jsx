import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { expect, describe, it, vi } from 'vitest';

import { Header } from './Header';
import authReducer, { logout } from '../store/auth/authSlice';

const renderWithStore = (preloadedState = { userData: { email: 'usuario@teste.com' } }) => {
  const store = configureStore({
    reducer: { userData: authReducer },
    preloadedState,
  });

  const spy = vi.spyOn(store, 'dispatch');

  render(
    <Provider store={store}>
      <Header />
    </Provider>,
  );

  return { store, spy };
};

describe('Header Component', () => {
  it('deve renderizar corretamente', () => {
    renderWithStore();
    expect(screen.getByText('ITA Frotas')).toBeInTheDocument();
  });

  it('deve exibir o email do usuÃ¡rio', () => {
    renderWithStore({ userData: { email: 'teste@email.com' } });
    expect(screen.getByText('teste@email.com')).toBeInTheDocument();
  });

  it('deve corresponder ao snapshot', () => {
    const { container } = renderWithStore();
    expect(container).toMatchSnapshot();
  });

  it('deve disparar logout ao clicar em "Sair"', () => {
    const { spy } = renderWithStore();

    fireEvent.click(screen.getByText('usuario@teste.com'));

    const logoutOption = screen.getByText('Sair');
    fireEvent.click(logoutOption);

    expect(spy).toHaveBeenCalledWith(logout());
  });
});
