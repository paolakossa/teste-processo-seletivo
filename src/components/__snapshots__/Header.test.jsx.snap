// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header Component > deve corresponder ao snapshot 1`] = `undefined`;
